### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'strip' failure even with no .condarc file

### ISSUE THREAD

**Issue by seibert** [conda/conda#3788]
I seem to have hit a variant of a bug reported elsewhere (#3783, #3774, #3768):

```
Rover:~ stan$ conda update numpy
Fetching package metadata .......
Solving package specifications: ..........
An unexpected error has occurred.
Please consider posting the following information to the
conda GitHub issue tracker at:

    https://github.com/conda/conda/issues



Current conda install:

               platform : osx-64
          conda version : 4.2.11
       conda is private : False
      conda-env version : 4.2.11
    conda-build version : 1.18.2
         python version : 2.7.12.final.0
       requests version : 2.11.1
       root environment : /Users/stan/anaconda  (writable)
    default environment : /Users/stan/anaconda
       envs directories : /Users/stan/anaconda/envs
          package cache : /Users/stan/anaconda/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : None
           offline mode : False



`$ /Users/stan/anaconda/bin/conda update numpy`




    Traceback (most recent call last):
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 479, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 145, in _main
        exit_code = args.func(args, p)
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/cli/main_update.py", line 65, in execute
        install(args, parser, 'update')
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/cli/install.py", line 308, in install
        update_deps=context.update_dependencies)
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/plan.py", line 526, in install_actions
        force=force, always_copy=always_copy)
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/plan.py", line 308, in ensure_linked_actions
        fetched_in = is_fetched(dist)
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/install.py", line 727, in is_fetched
        for fn in package_cache().get(dist, {}).get('files', ()):
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/install.py", line 675, in package_cache
        add_cached_package(pdir, url)
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/install.py", line 633, in add_cached_package
        schannel = Channel(url).canonical_name
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/models/channel.py", line 161, in __call__
        c = Channel.from_value(value)
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/models/channel.py", line 211, in from_value
        return Channel.from_url(value)
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/models/channel.py", line 196, in from_url
        return parse_conda_channel_url(url)
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/models/channel.py", line 132, in parse_conda_channel_url
        configured_token) = _read_channel_configuration(scheme, host, port, path)
      File "/Users/stan/anaconda/lib/python2.7/site-packages/conda/models/channel.py", line 122, in _read_channel_configuration
        return (Url(host=host, port=port).url.rstrip('/'), path.strip('/') or None,
    AttributeError: 'NoneType' object has no attribute 'strip'
```

However, unlike those reports, I don't have a `.condarc` file to modify.  Any ideas how to fix?

**Comment by seibert** [comment]
(BTW, this started happening as soon as I accidentally updated from conda 4.1 to 4.2)

**Comment by seibert** [comment]
Follow-up: I added some print logging to conda, and it appears to be blowing up when trying to parse the following channel URL: `http://kts02:8000/osx-64/tradesim-2.2.6098-py27_0.tar.bz2`

This is an old package I installed somewhere, but I'm not sure where this URL is being picked up from, so I can remove it.

**Comment by seibert** [comment]
OK, it appears to have been caused by that URL appearing in `$HOME/anaconda/pkgs/urls.txt`.  I moved that file aside, and conda was able to operate normally.

What is `urls.txt` for?

**Comment by kalefranz** [comment]
@seibert We released a conda patch version 4.2.12 yesterday. Can you update and see if the problem persists? If so, can you give me the new stack trace? It's possible the line numbers have changed.

**Comment by seibert** [comment]
Looks like that fixed it.  Thanks!
